Lockbox for Decentralized Grants Allocation (#858)
zcash/zips: zip-draft-XXXX.rst

::

  ZIP: Unassigned
  Title: Lockbox for Decentralized Grants Allocation
  Owners: Skylar Saveland <skylar@free2z.com>
          Jason McGee <>
  Credits:
  Status: Draft
  Category: Consensus
  Created: 2024-06-26
  License: MIT
  Pull-Request: <https://github.com/zcash/zips/pull/858>

Terminology
===========

The key words "MUST", "REQUIRED", "MUST NOT", "SHOULD", and "MAY" in this
document are to be interpreted as described in BCP 14 [#BCP14]_ when, and
only when, they appear in all capitals.

.. {Avoid duplicating definitions from other ZIPs. Instead use wording like this:}

.. The terms "Mainnet" and "Testnet" in this document are to be interpreted as
.. defined in the Zcash protocol specification [#protocol-networks]_.

.. The term "full validator" in this document is to be interpreted as defined in
.. the Zcash protocol specification [#protocol-blockchain]_.

.. The terms below are to be interpreted as follows:

.. {Term to be defined}
..   {Definition.}
.. {Another term}
..   {Definition.}

.. |percentage| replace:: 50%

Abstract
========

This ZIP proposes the allocation of |percentage| of the Zcash block reward,

> post-November 2024 halving, to a decentralized "lockbox." Currently, 80% of the

Reviewer: ```suggestion
post-November 2024 halving, to an in-protocol "lockbox." Currently, 80% of the
```
The lockbox itself is a feature of the protocol; the ZIP should identify that the funds 
stored in the lockbox are intended to be distributed in the future to fund development
via a to-be-determined decentralized protocol.

block reward goes to miners, while 20% is distributed among the Major Grants
Fund, Electric Coin Company (ECC), and the Zcash Foundation (ZF). If no changes
are made, this 20% dev fund will expire, resulting in the entire block reward
going to miners, leaving no block-reward funds for essential protocol
development, security, marketing, or legal expenses.

The proposed lockbox addresses significant issues observed with [#zip-1014]_,
such as regulatory risks, inefficiencies in funding organizations instead of

> projects, tax compliance inefficiencies, and centralization. By disbursing

Reviewer: A future protocol for disbursing these funds should take these issues into account, but
since the current ZIP doesn't propose the disbursement mechanism, we don't know yet
whether it will actually satisfy those regulatory concerns. You mention future disbursement in the motivation section, but the TBD nature of that should be noted in the abstract.

the remaining 20% distributed among the Major Grants Fund (8%), Electric Coin
Company (ECC) (7%), and the Zcash Foundation (ZF) (5%). This funding structure
supports various essential activities such as protocol development, security,
marketing, and legal expenses. However, this model will expire in November
2024, leading to the entire block reward being allocated to miners if no
changes are made. This expiration poses several significant issues:

1. **Regulatory Risks**: The current model, as observed with ZIP-1014, involves
   direct funding of US-based organizations. This can potentially attract
   regulatory scrutiny from entities such as the SEC, posing legal risks to the
   Zcash ecosystem.

2. **Funding Inefficiencies**: Directly funding organizations rather than
   specific projects leads to inefficiencies. Organizations may accumulate
   excess funds while specific projects may not receive adequate resources. A
   more granular approach, funding projects directly, would be more efficient.

3. **Tax Compliance Complications**: The current model introduces tax
   inefficiencies, as 501(c)(3) organizations must treat received block rewards

>    as income, leading to significant tax liabilities. This requires selling

Reviewer: This is likely to be the case for any entity that ultimately receives the block rewards in the future.

Author: entity or individual yes; but perhaps not if it's one multisig to another? Eventually an entity or individual will have some tax liability but this proposal could put it farther away from the protocol layer I think.

   some of the block rewards to cover these taxes, which is financially
   inefficient.

4. **Centralization Concerns**: The existing funding model centralizes
   decision-making power within a few organizations, which contradicts the
   decentralized ethos of blockchain technology. Traditional organizational
   structures with boards and executives introduce a single point of failure

>    and limit community involvement in funding decisions.

Reviewer: This is _kind of_ ok since it's in motivation, but it also assumes something about the disbursement that we don't really know yet.

   the best methods for handling taxable events related to the disbursement of
   funds.

6. **Impact Assessment**: The long-term impact of reallocating a portion of the
   block reward to the lockbox on the Zcash ecosystem, including its effect on
   miners, developers, and the broader community, is not analyzed in this ZIP.
   Subsequent proposals will need to evaluate the outcomes and make necessary
   adjustments based on real-world feedback and data.

By focusing on the establishment of the lockbox, this ZIP aims to lay the
groundwork for a more decentralized and efficient funding mechanism. The
deferred issues will be crucial in shaping the final implementation and
ensuring the proposed system's success and sustainability.

Specification
=============

This section describes the consensus rules and protocol changes required to
implement the proposed in-protocol lockbox for deferred development fund
allocation. The specification outlines the creation, management, and
integration of the lockbox within the Zcash protocol.

Deferred Development Fund Chain Value Pool Balance
--------------------------------------------------

> A new deferred development fund chain value pool balance will be created. This

Reviewer: ```suggestion
A new development fund value pool balance, denominated in zatoshis, will be created. This
```
"deferred" probably isn't necessary here; it's deferred given that no disbursement mechanism is specified, but once such a disbursement mechanism is added by a future ZIP we don't want it to be referred to as "deferred" any more, so it should probably just be a development funding pool. This is kind of a naming nit, but if this is the funding option that is selected we'll want the node implementations to be forward-looking when they name variables and so forth.

balance will be initialized to zero at the NU6 activation block.

Coinbase Transaction Adjustments
--------------------------------

For coinbase transactions, a specified proportion of the block subsidy will be
added to the deferred development fund chain value pool balance and subtracted

> from the transparent transaction value pool. The exact proportions

Reviewer: This subtraction will occur _after_ block reward value is logically added to the transparent value pool.

(numerator/denominator) of the block subsidy allocated to the deferred
development fund must be defined to ensure the correct allocation at every
block.

Proportion Specification
------------------------

The proportions of the block subsidy allocated to the deferred development fund
and any existing funding streams must add up to the intended total at every
block. Before NU6 activation, the actual percentage allocated to the deferred
development fund will be zero. After NU6 activation, this percentage will be

> set according to the proposal.

Reviewer: This should be explicit, instead of "set according to the proposal". It's okay for this to be redundant with respect to other parts of the spec.

Backwards Compatibility
-----------------------

Full validators can implement the reserve consensus rules without NU gating.
Prior to NU6 activation, the percentage allocated to the reserve pool can be

> set to zero or be defined as an empty set, similar to existing funding streams.

Reviewer: Another implementation option here would be to define a new type of funding stream that is "direct this value to the in-protocol development fund instead of constructing a UTXO."

Author: you can check the latest revision to see if it makes sense.

   design should ensure compliance with applicable laws and regulations to
   support the long-term sustainability of the funding model.

4. **Efficient Allocation Preparation**: The lockbox must be prepared to
   allocate resources efficiently once the disbursement mechanism is defined.
   This includes ensuring that funds are readily available for future projects
   and not tied up in organizational overhead.

5. **Potential Tax Efficiency**: The lockbox should be designed with the
   potential to defer taxable events until the liquidation of funds by
   recipients, allowing for more flexibility in handling tax liabilities
   compared to direct funding of US-based organizations.

6. **Security of Funds**: The lockbox must implement robust security measures
   to protect against unauthorized access or misuse of funds. This includes
   ensuring the integrity of the future decentralized voting/signature
   mechanism.

7. **Future Adaptability**: The lockbox design should be flexible and adaptable
   to accommodate the disbursement mechanism defined in a future ZIP. This
   ensures that the system can evolve based on feedback and new developments in

>    the Zcash ecosystem.

Reviewer 2: I'm not sure these are _requirements_, since they couldn't possibly be fully addressed by the lockbox itself. I think it's fine to leave them in the motivation.

   addressed in a future ZIP. This includes specifics on how funds will be
   allocated, the voting or decision-making process, and the structure of the
   decentralized mechanism (such as a DAO).

2. **Specific Allocation Percentages**: While this ZIP proposes the allocation
   of |percentage| of the block reward to the lockbox, it does not prescribe
   the detailed percentages for various funding categories or the specific
   amounts that will be distributed to different projects or entities.

3. **Regulatory Compliance Details**: The proposal outlines the potential to
   reduce regulatory risks by avoiding direct funding of US-based
   organizations, but it does not detail specific regulatory compliance
   strategies. Future ZIPs will need to address how the disbursement mechanism
   complies with applicable laws and regulations.

4. **Technical Implementation**: The technical specifics of how the lockbox
   will be integrated into the Zcash protocol, including any necessary changes
   to the codebase and the precise methods for ensuring security and
   transparency, are not covered in this ZIP. These details will be provided in

>    subsequent proposals and technical documentation.

Reviewer 2: I don't think it's possible to leave this out - if the ZIP is approved it will have to be implemented and we'll need this information regardless. But note that ZIPs don't have to explain how things are going to be implemented, just what is the change in the protocol that needs to be implemented (i.e. what you have already written in the Specification section, so my suggestion is to simply remove this item)

   miners, developers, and the broader community, is not analyzed in this ZIP.
   Subsequent proposals will need to evaluate the outcomes and make necessary
   adjustments based on real-world feedback and data.

By focusing on the establishment of the lockbox, this ZIP aims to lay the
groundwork for a more decentralized and efficient funding mechanism. The
deferred issues will be crucial in shaping the final implementation and
ensuring the proposed system's success and sustainability.

Specification
=============

This section describes the consensus rules and protocol changes required to
implement the proposed in-protocol lockbox for deferred development fund
allocation. The specification outlines the creation, management, and
integration of the lockbox within the Zcash protocol.

Deferred Development Fund Chain Value Pool Balance
--------------------------------------------------

A new development fund value pool balance, denominated in zatoshis, will be
created. This balance will be initialized to zero at the NU6 activation block.

Coinbase Transaction Adjustments
--------------------------------

For coinbase transactions, a specified proportion of the block subsidy will be

> added to the deferred development fund chain value pool balance and subtracted

Reviewer 2: I suggest using capitals like "Deferred Development Fund" to make it clear it's "the thing" being proposed. You could then add it to the Terminology section.

from the transparent transaction value pool. This subtraction will occur after
block reward value is logically added to the transparent value pool. The exact
proportions (numerator/denominator) of the block subsidy allocated to the
deferred development fund must be defined to ensure the correct allocation at
every block.

Proportion Specification
------------------------

> The proportions of the block subsidy allocated to the deferred development fund
> and any existing funding streams must add up to the intended total at every
> block. Before NU6 activation, the actual percentage allocated to the deferred
> development fund will be zero. After NU6 activation, |percentage| of the block
> reward will be allocated to the deferred development fund.

Reviewer 3: This was fine as a comment in the HackMD it was taken from, but is confusing here (as this ZIP draft does not propose for any funding streams to be concurrent with the lockbox). Something like it may end up in the ZIP that defines how the core lockbox / issuance pool mechanism works, but for today's deadline, remove all but the last sentence:
```suggestion
After NU6 activation, |percentage| of the block reward will be allocated to the
deferred development fund.
```

::

  ZIP: Unassigned
  Title: Lockbox for Decentralized Grants Allocation
  Owners: Skylar Saveland <skylar@free2z.com>
          Jason McGee <>
  Credits: @daira
           @str4d
           @nuttycom
  Status: Draft
  Category: Consensus
  Created: 2024-06-26
  License: MIT
  Pull-Request: <https://github.com/zcash/zips/pull/858>

Terminology
===========

> The key words "MUST", "REQUIRED", "MUST NOT", "SHOULD", and "MAY" in this

Author: TODO: do a pass for when these MUST be capitalized ;9

Author: I'm pretty happy with it, I think. Some things we might want to look at:

1. replace the comment about terminology with any actual terminology we should define
2. Capitalize "MUST", "REQUIRED", "MUST NOT", "SHOULD", and "MAY" where it makes sense
3. Improve the references (?)

In addition, the `Specification`  section might need some more revisions? Does it need to be more specific and more technical or is it good enough?

Author: Maybe in `non-requirements` it could be mentioned that this proposal could sit alongside other things like "ZSF" and some form of continuation of the "direct streams" current funding model (eg some percentage to ZCG?). I think the "lockbox" ZIP as written doesn't preclude other things that could be done with the block reward?

Reviewer 4: I think another primary motivation is that the lockbox serves an important purpose if the community cannot come to consensus on a dev fund proposal or if there is no support for a direct funding model. By allocating 20% of the block rewards to the lockbox, the dev fund can continue to accrue. Without this, the 20% would go to miners, and redirecting these rewards from miners to a new Dev Fund later would require a future network upgrade, which could be controversial.

Reviewer 4: > Maybe in `non-requirements` it could be mentioned that this proposal could sit alongside other things like "ZSF" and some form of continuation of the "direct streams" current funding model (eg some percentage to ZCG?). I think this "lockbox" ZIP as written doesn't preclude other things that could be done with the block reward?

Re the ZSF: It's worth noting that the lockbox and ZSF are compatible. As Daira points out in hir writeup, the deferred funds held in the lockbox are part of circulating supply, whereas the ZSF contains unissued supply. Ze says, "mixing these two is a bad idea." Also, the primary (or sole) use case for the lockbox is to temporarily serve as a reserve for the dev fund until the community decides how it should be allocated. In contrast, the ZSF has a number of different long-term use cases.

Re: "some percentage to ZCG": Yes, it makes sense to reference a ZCG allocation ini the non-requirements section. I mentioned this to strad in the ZIP Editor office hours and said for a hybrid version where a portion goes to the lockbox and a portion to ZCG, we can use ZIP 207 as a template to define the funding stream to ZCG.  I'll take a look at that later today and will follow up with you.

Reviewer 5: Just to clarify, the write-up was written jointly by me, @str4d, and @nuttycom .

Requirements
============

1. **Decentralized Funding Mechanism**: The new funding model must implement a
   decentralized mechanism for disbursing funds. This mechanism should resemble
   a DAO, ensuring that the community has a direct say in funding decisions.
   This is crucial to avoid centralization and ensure transparency and fairness

>    in the allocation of resources.

Reviewer: This is a requirement for the future disbursement mechanism; the requirements section of this ZIP should be targeted to the protocol changes that _this_ ZIP will require.

2. **Regulatory Compliance**: The proposed funding model must be designed to
   minimize regulatory risks. This involves avoiding direct funding of US-based
   organizations to prevent potential scrutiny from regulatory bodies such as
   the SEC. The mechanism should comply with applicable laws and regulations to
   ensure the long-term sustainability of the funding model.

3. **Efficiency in Funding Allocation**: The new model should focus on funding
   specific projects rather than organizations. This approach aims to ensure
   that resources are utilized efficiently and that projects receive the
   necessary support without accumulating excess funds in organizations.

4. **Simplified Tax Compliance**: The funding mechanism must simplify tax
   compliance for recipients. This can be achieved by deferring tax liabilities
   to the recipients, allowing them to handle taxes according to their
   circumstances, thus avoiding the need for organizations to sell block

>    rewards to cover taxes.

Reviewer: This is going to be relatively complicated from a legal perspective, and it might not actually result in tax savings depending upon how the entities that ultimately receive the funds are structured - for example, if they're for-profit corporations, it could _increase_ the tax burden. It's unlikely that individual contributors can be readily compensated in this fashion, both due to administrative overhead and because (at least in the U.S.) even an individual contributor would likely need to incorporate as a single-member LLC for liability reasons.

Author: Good point, requirements need to be rewritten from the standpoint that this is just to create the "lockbox" and not implement the disbursement mechanism anyways ... new draft incoming ...

5. **Community Involvement**: The funding model must enhance community
   involvement in decision-making processes. This includes mechanisms for the
   community to propose, discuss, and vote on projects that should receive
   funding. Ensuring broad community participation is key to aligning funded
   projects with the community's priorities.

6. **Security of Funds**: The mechanism for storing and disbursing funds must
   be secure. This includes protecting against key compromise, ensuring the
   integrity of the decentralized voting/signature mechanism, and implementing

>    robust security measures to prevent unauthorized access or misuse of funds.

Reviewer: This is applicable to the current ZIP independent of how the future ZIP does disbursement (the future ZIP would also need this kind of stanza, but it would then directly address questions of security.)
